accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,447 @@
+It does this by scanning the `PATH` environment variable for plugins whose file name begins with `nuget-plugin-`.
+On Windows, NuGet will look for plugins with a `.exe` or `.bat` extension, whereas on Linux/Mac, it will look for plugins with the executable bit set.
+These plugins are launched in a separate process, which aligns with the current design.
+
+## Motivation
+
+Currently, the NuGet plugin architecture requires support and deployment of multiple versions.
+For `.NET Framework`, NuGet searches for files ending in `*.exe`, while for `.NET Core`, it searches for files ending in `*.dll`.
+These files are stored in two distinct folders, `netfx` and `netcore`, under the NuGet plugins base path.
+This design decision is due to the different entry points for `.NET Core` and `.NET Framework`, as explained in the [original design](https://github.com/NuGet/Home/wiki/NuGet-cross-plat-authentication-plugin#plugin-installation-and-discovery).
+`.NET Core` uses files with a `dll` extension, while `.NET Framework` uses files with an `exe` extension as entry points.
+This distinction is further highlighted by the two plugin path environment variables, `NUGET_NETFX_PLUGIN_PATHS` and `NUGET_NETCORE_PLUGIN_PATHS`, which must be set for each framework type.
+
+Another motivating factor for this work is the current story for installing these credential providers, which is not ideal.
+For instance, let's consider the two cross-platform authentication plugins that I am aware of while writing this specification:
+
+1. **Azure Artifacts Credential Provider** - The [setup](https://github.com/microsoft/artifacts-credprovider/tree/master?tab=readme-ov-file#setup) instructions vary based on the platform.
+2. **AWS CodeArtifact Credential Provider** - The AWS team has developed a [.NET tool](https://docs.aws.amazon.com/codeartifact/latest/ug/nuget-cli.html#nuget-configure-cli) to facilitate authentication with their private NuGet feeds.
+In their current implementation, they've added a subcommand, `codeartifact-creds install`, which copies the credential provider to the NuGet plugins folder.
+
+NuGet plugin authors could leverage the .NET SDK to allow their customers to install, uninstall, or update the plugins more efficiently.
+
+## Explanation
+
+### Functional explanation
+
+A deployment solution for .NET is the [.NET tools](https://learn.microsoft.com/dotnet/core/tools).
+These tools provide a seamless installation and management experience for NuGet packages in the .NET ecosystem.
+The use of .NET tools as a deployment mechanism has been a recurring request from users and internal partners who need to authenticate with private repositories.
+Currently, this solution works for the Windows .NET Framework.
+However, the goal is to extend this support cross-platform for all supported .NET runtimes.
+
+The reasons why `.NET tools` were chosen as the deployment mechanism are mentioned below:
```

Comment:
nit: This would be one of the recommended mechanisms right? 
Cause technically we're going to be more generic at launch time.

Reply:
Changed the wording to `The reasons why .NET tools were chosen as the recommended deployment mechanism are mentioned below:`